core/src/main/java/org/infinispan/container/entries/ReadCommittedEntry.java

```diff
@@ -194,7 +195,9 @@ public final void commit(DataContainer container, EntryVersion newVersion) {
             if (isRemoved() && !isEvicted()) ahm.markRemoved(true);
          }
 
-         if (isRemoved()) {
+         if (isEvicted() && container instanceof DefaultDataContainer) {
```

Comment:
This is another API issue.  I can't add the evict method to the DataContainer interface.  I had debated about adding another interface, which I can do if we want so if someone implements their own container they could implement it ?